presto-main/src/main/java/com/facebook/presto/server/CoordinatorModule.java

```diff
@@ -175,14 +171,8 @@ protected void setup(Binder binder)
         binder.bind(InternalResourceGroupManager.class).in(Scopes.SINGLETON);
         newExporter(binder).export(InternalResourceGroupManager.class).withGeneratedName();
         binder.bind(ResourceGroupManager.class).to(InternalResourceGroupManager.class);
-        binder.bind(LegacyResourceGroupConfigurationManagerFactory.class).in(Scopes.SINGLETON);
-        if (buildConfigObject(FeaturesConfig.class).isResourceGroupsEnabled()) {
-            binder.bind(QueryQueueManager.class).to(InternalResourceGroupManager.class);
-        }
-        else {
-            binder.bind(QueryQueueManager.class).to(SqlQueryQueueManager.class).in(Scopes.SINGLETON);
-            binder.bind(new TypeLiteral<List<QueryQueueRule>>() {}).toProvider(QueryQueueRuleFactory.class).in(Scopes.SINGLETON);
-        }
+        binder.bind(LegacyResourceGroupConfigurationManager.class).in(Scopes.SINGLETON);
```

Comment:
Why do we need a legacy manager at all?

Reply:
It is used by `InternalResourceGroupManager` when there is no resource group configuration. I agree it should be removed, but that's a separate project (I didn't see an easy way).